singularity-eos/base/root-finding-1d/root_finding.hpp

```diff
@@ -235,6 +237,71 @@ PORTABLE_INLINE_FUNCTION Status regula_falsi(const T &f, const Real ytarget,
   return status;
 }
 
+// solves for f(x,params) - ytarget = 0
+template <typename T>
+PORTABLE_INLINE_FUNCTION Status newton_raphson(const T &f, const Real ytarget,
+                                               const Real guess, const Real a,
+                                               const Real b, const Real ytol, Real &xroot,
+                                               const RootCounts *counts = nullptr,
+                                               const bool &verbose = false) {
```

Comment:
I think it's okay to use the same function signature as the regula falsi method, but I think you should at least add a comment about the behavior of the callable `f` being different (i.e. needing to return both function and derivative). Anybody using it now will quickly run into a compile time error if they try to just return the function, but why not save them the work of compiling to figure that out?

In the future I could see possibly wanting to develop an overload for this signature that uses the same behavior for `f` as the regula falsi and then wraps that callable in a function to perform a finite difference of some sort when an analytic derivative isn't available. Regardless of whether that happens though, comments will always help in understanding the subtle differences in template behavior between the root finding methods.

Reply:
I agree with @jhp-lanl here